spec/models/workflow_spec.rb

```diff
@@ -145,6 +145,45 @@
     end
   end
 
+  describe '#retire_subject' do
+    it 'marks a counted subject as retired' do
+      workflow = create(:workflow_with_subject_sets)
+      subject = create(:subject, subject_sets: workflow.subject_sets)
+      count = create(:subject_workflow_count, set_member_subject: subject.set_member_subjects.first, workflow: workflow)
+
+      workflow.retire_subject(subject.id)
+      expect(subject.retired_for_workflow?(workflow)).to be_truthy
+      expect(SubjectWorkflowCount.retired.count).to eq(2)
+    end
+
+    it 'marks an uncounted subject as retired' do
+      workflow = create(:workflow_with_subject_sets)
+      subject = create(:subject, subject_sets: workflow.subject_sets)
+
+      workflow.retire_subject(subject.id)
+      expect(subject.retired_for_workflow?(workflow)).to be_truthy
+      expect(SubjectWorkflowCount.retired.count).to eq(2)
+    end
+
+    it 'leaves a retired subject retired' do
+      workflow = create(:workflow_with_subject_sets)
+      subject = create(:subject, subject_sets: workflow.subject_sets)
+
+      workflow.retire_subject(subject.id)
+      workflow.retire_subject(subject.id)
```

Comment:
cool want to add an `aggregate_failures` double expectation that the retire at stamp doesn't change?